### PR TITLE
PAPP-12188 : Mattermost feature request

### PR DESCRIPTION
**Pull request by pzhou-splunk**
- Earlier only public channels were allowed to do communication, now private channels are also allowed. Affected actions are send message, list posts, list channels, upload file. ref : [PAPP-12188](https://splunk.atlassian.net/browse/PAPP-12188)
- Added missing data paths
- Resolved formatting related issues

### CODE REVIEW

**Review by ngaglani-crest — DISMISSED** [review]

**Review by ngaglani-crest — DISMISSED** [review]

**Comment by paulnawat** [comment]
Could you add some description for this PR? What are the goals of this PR?

**Review by brian-phantom — DISMISSED** [review]

File: mattermost_connector.py [hunk 1]
```diff
@@ -1659,10 +1705,12 @@ def finalize(self):
 
             headers = dict()
             headers['Cookie'] = 'csrftoken={}'.format(csrftoken)
-            headers['Referer'] = 'https://127.0.0.1/login'
+            headers['Referer'] = BaseConnector._get_phantom_base_url() + \
+                "login"
```
Comment (paulnawat) [inline mattermost_connector.py]: Is this really exceeded the character limit for the line? It looks like it should fit. 😅

File: mattermost_connector.py [hunk 2]
```diff
@@ -1502,7 +1540,8 @@ def _handle_list_channels(self, param):
         :return: status(phantom.APP_SUCCESS/phantom.APP_ERROR)
         """
 
-        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        self.save_progress("In action handler for: {0}".format(
+            self.get_action_identifier()))
```
Comment (paulnawat) [inline mattermost_connector.py]: nit: Can this fit in one line?

File: mattermost_connector.py [hunk 3]
```diff
@@ -987,50 +1010,39 @@ def _verify_channel(self, action_result, team_id, channel):
...
-                            get('name').lower():
-                        channel_id = each_channel.get('id')
-                        return phantom.APP_SUCCESS, channel_id
+        # For any action other than list channel
+        if not self.get_action_identifier() == 'list_channels':
+            # Fetch Channel ID from Channel name
+            for each_channel in response_json:
+                # Check if either channel name or channel ID matches
+                if channel.lower() == each_channel.get('id').lower() or channel.lower() == each_channel.\
+                        get('name').lower():
```
Comment (paulnawat) [inline mattermost_connector.py]: suggestion: Should we use bracket to wrap multi line condition instead?
```python
if (channel.lower() == each_channel.get('id').lower() or 
                channel.lower() == each_channel.get('name').lower()):
         pass
```

**Review by paulnawat — DISMISSED** [review]
Changes look great. Thanks!
I left a few minor suggestions: If the code can fit in one line, I think it should be kept the same.

**Review by gary-phantom — DISMISSED** [review]

**Comment by ngaglani-crest** [comment]
> Changes look great. Thanks! I left a few minor suggestions: If the code can fit in one line, I think it should be kept the same.

@paulnawat, We had changed these lines because while committing the code, flake8 hook from pre-commit failed with the below message:

image

For now, we have disabled pre-commit on local and committed the changes.

It looks like the hook is using `max-line-length = 130` instead of `145`.

**Review by ngaglani-crest — APPROVED** [review]

File: mattermost_connector.py (hunk 1 above)
Comment (ngaglani-crest) [reply mattermost_connector.py]: Done.

File: mattermost_connector.py (hunk 2 above)
Comment (ngaglani-crest) [reply mattermost_connector.py]: Done,

File: mattermost_connector.py (hunk 3 above)
Comment (ngaglani-crest) [reply mattermost_connector.py]: We have kept both these conditions at the same line, as total characters at the line is <145.

**Comment by ngaglani-crest** [comment]
Merging this PR as it was earlier approved by Gary and Paul. Also, the suggested changes have been updated in the code.

**Pull request merged**